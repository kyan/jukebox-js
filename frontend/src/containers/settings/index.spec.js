import React from 'react'
import { shallow, mount } from 'enzyme'
import configureMockStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import SettingsCont, { Settings } from './index'

describe('CurrentTrackContainer', () => {
  let wrapper
  const mockDispatch = jest.fn()

  describe('render just the app without redux', () => {
    beforeEach(() => {
      wrapper = shallow(
        <Settings
          dispatch={mockDispatch}
          settings={{ open: false }}
        />
      )
    })

    it('renders as expected', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('renders the modal', () => {
      wrapper.find('.jb-settings-toggle').simulate('click')
      expect(mockDispatch.mock.calls.length).toEqual(1)
      expect(mockDispatch.mock.calls[0][0]).toEqual({
        type: 'actionToggleSettings'
      })
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('render the settings modal open', () => {
    const store = configureMockStore()({
      settings: { open: true }
    })

    it('renders as expected', () => {
      wrapper = mount(
        <Provider store={store}>
          <SettingsCont />
        </Provider>
      )

      expect(wrapper).toMatchSnapshot()
    })
  })
})
