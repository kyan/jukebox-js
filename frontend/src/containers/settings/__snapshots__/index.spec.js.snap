// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentTrackContainer just the app without redux when authorised user renders as expected 1`] = `
<React.Fragment>
  <Image
    as="img"
    className="jb-settings-toggle"
    floated="right"
    onClick={[Function]}
    rounded={true}
    size="mini"
    src="https://www.gravatar.com/avatar/hash123"
    title="John Doe"
    ui={true}
  />
  <Modal
    closeIcon={true}
    closeOnDimmerClick={true}
    closeOnDocumentClick={false}
    dimmer="blurring"
    eventPool="Modal"
    onClose={[Function]}
    open={false}
    size="small"
  >
    <ModalHeader>
      Settings
    </ModalHeader>
    <ModalContent>
      <ModalDescription>
        <Form
          as="form"
        >
          <FormInput
            action={
              Object {
                "className": "jb-authenticate-button",
                "content": "Authenticated",
                "icon": "checkmark",
                "labelPosition": "right",
                "onClick": [Function],
              }
            }
            as={[Function]}
            control={[Function]}
            icon="user"
            iconPosition="left"
            label="Enter your USERNAME below"
            onChange={[Function]}
            placeholder="example: user123"
            size="massive"
            value="user123"
          />
          <Progress
            percent={100}
            success={true}
          >
            user123 is now logged in
          </Progress>
        </Form>
      </ModalDescription>
    </ModalContent>
  </Modal>
</React.Fragment>
`;

exports[`CurrentTrackContainer just the app without redux when authorised user renders the modal 1`] = `
<React.Fragment>
  <Image
    as="img"
    className="jb-settings-toggle"
    floated="right"
    onClick={[Function]}
    rounded={true}
    size="mini"
    src="https://www.gravatar.com/avatar/hash123"
    title="John Doe"
    ui={true}
  />
  <Modal
    closeIcon={true}
    closeOnDimmerClick={true}
    closeOnDocumentClick={false}
    dimmer="blurring"
    eventPool="Modal"
    onClose={[Function]}
    open={false}
    size="small"
  >
    <ModalHeader>
      Settings
    </ModalHeader>
    <ModalContent>
      <ModalDescription>
        <Form
          as="form"
        >
          <FormInput
            action={
              Object {
                "className": "jb-authenticate-button",
                "content": "Authenticated",
                "icon": "checkmark",
                "labelPosition": "right",
                "onClick": [Function],
              }
            }
            as={[Function]}
            control={[Function]}
            icon="user"
            iconPosition="left"
            label="Enter your USERNAME below"
            onChange={[Function]}
            placeholder="example: user123"
            size="massive"
            value="user123"
          />
          <Progress
            percent={100}
            success={true}
          >
            user123 is now logged in
          </Progress>
        </Form>
      </ModalDescription>
    </ModalContent>
  </Modal>
</React.Fragment>
`;

exports[`CurrentTrackContainer just the app without redux when nonauthorised user renders as expected 1`] = `
<React.Fragment>
  <Button
    as="button"
    className="jb-settings-toggle"
    floated="right"
    icon="user"
    onClick={[Function]}
  />
  <Modal
    closeIcon={true}
    closeOnDimmerClick={true}
    closeOnDocumentClick={false}
    dimmer="blurring"
    eventPool="Modal"
    onClose={[Function]}
    open={false}
    size="small"
  >
    <ModalHeader>
      Settings
    </ModalHeader>
    <ModalContent>
      <ModalDescription>
        <Form
          as="form"
        >
          <FormInput
            action={
              Object {
                "className": "jb-authenticate-button",
                "content": "Click to validate",
                "icon": "checkmark",
                "labelPosition": "right",
                "onClick": [Function],
              }
            }
            as={[Function]}
            control={[Function]}
            icon="user"
            iconPosition="left"
            label="Enter your USERNAME below"
            onChange={[Function]}
            placeholder="example: user123"
            size="massive"
            value=""
          />
          <Progress
            error={true}
            percent={100}
          >
            Unknown User!
          </Progress>
        </Form>
      </ModalDescription>
    </ModalContent>
  </Modal>
</React.Fragment>
`;

exports[`CurrentTrackContainer render the settings modal open renders as expected 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Component)>
    <Component
      dispatch={[Function]}
      settings={
        Object {
          "open": true,
        }
      }
    >
      <Button
        as="button"
        className="jb-settings-toggle"
        floated="right"
        icon="user"
        onClick={[Function]}
      >
        <button
          className="ui icon right floated button jb-settings-toggle"
          onClick={[Function]}
          role="button"
        >
          <Icon
            as="i"
            key="user"
            name="user"
          >
            <i
              aria-hidden="true"
              className="user icon"
            />
          </Icon>
        </button>
      </Button>
      <Modal
        closeIcon={true}
        closeOnDimmerClick={true}
        closeOnDocumentClick={false}
        dimmer="blurring"
        eventPool="Modal"
        onClose={[Function]}
        open={true}
        size="small"
      >
        <Portal
          className="ui page modals dimmer transition visible active"
          closeOnDocumentClick={false}
          closeOnEscape={true}
          closeOnRootNodeClick={true}
          eventPool="Modal"
          mountNode={
            <body
              class="dimmable dimmed blurring"
            >
              <div
                class="ui page modals dimmer transition visible active"
              >
                <div
                  class="ui small modal transition visible active"
                >
                  <i
                    aria-hidden="true"
                    class="close icon"
                  />
                  <div
                    class="header"
                  >
                    Settings
                  </div>
                  <div
                    class="content"
                  >
                    <div
                      class="description"
                    >
                      <form
                        class="ui form"
                      >
                        <div
                          class="field"
                        >
                          <label>
                            Enter your USERNAME below
                          </label>
                          <div
                            class="ui massive action left icon input"
                          >
                            <input
                              placeholder="example: user123"
                              type="text"
                              value=""
                            />
                            <button
                              class="ui icon right labeled button jb-authenticate-button"
                              role="button"
                            >
                              <i
                                aria-hidden="true"
                                class="checkmark icon"
                              />
                              Click to validate
                            </button>
                            <i
                              aria-hidden="true"
                              class="user icon"
                            />
                          </div>
                        </div>
                        <div
                          class="ui error progress"
                          data-percent="100"
                        >
                          <div
                            class="bar"
                            style="width: 100%;"
                          />
                          <div
                            class="label"
                          >
                            Unknown User!
                          </div>
                        </div>
                      </form>
                    </div>
                  </div>
                </div>
              </div>
            </body>
          }
          onClose={[Function]}
          onMount={[Function]}
          onOpen={[Function]}
          onUnmount={[Function]}
          open={true}
          openOnTriggerClick={true}
        />
      </Modal>
    </Component>
  </Connect(Component)>
</Provider>
`;
